### ISSUE TITLE
Getting started guide don't work

### ISSUE THREAD

**Issue by mkboel** [ansible/ansible-container#319]
##### ISSUE TYPE

 - Documentation Report

##### OS / ENVIRONMENT

```
Ansible Container, version 0.2.0
Linux, fs0f642, 4.4.0-45-generic, #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016, x86_64
2.7.11+ (default, Apr 17 2016, 14:00:29) 
[GCC 5.3.1 20160413] /usr/bin/python
No DOCKER_HOST environment variable found. Assuming UNIX socket at /var/run/docker.sock
{u'Architecture': u'x86_64',
 u'BridgeNfIp6tables': True,
 u'BridgeNfIptables': True,
 u'CPUSet': True,
 u'CPUShares': True,
 u'CgroupDriver': u'cgroupfs',
 u'ClusterAdvertise': u'',
 u'ClusterStore': u'',
 u'Containers': 8,
 u'ContainersPaused': 0,
 u'ContainersRunning': 1,
 u'ContainersStopped': 7,
 u'CpuCfsPeriod': True,
 u'CpuCfsQuota': True,
 u'Debug': False,
 u'DefaultRuntime': u'runc',
 u'DockerRootDir': u'/var/lib/docker',
 u'Driver': u'aufs',
 u'DriverStatus': [[u'Root Dir', u'/var/lib/docker/aufs'],
                   [u'Backing Filesystem', u'extfs'],
                   [u'Dirs', u'53'],
                   [u'Dirperm1 Supported', u'true']],
 u'ExecutionDriver': u'',
 u'ExperimentalBuild': False,
 u'HttpProxy': u'',
 u'HttpsProxy': u'',
 u'ID': u'XLSS:FYXO:3ONT:FV7C:R56O:2W5E:K4HK:UAKF:JL6R:6MJW:322N:RBUR',
 u'IPv4Forwarding': True,
 u'Images': 26,
 u'IndexServerAddress': u'https://index.docker.io/v1/',
 u'KernelMemory': True,
 u'KernelVersion': u'4.4.0-45-generic',
 u'Labels': None,
 u'LiveRestoreEnabled': False,
 u'LoggingDriver': u'json-file',
 u'MemTotal': 33659101184,
 u'MemoryLimit': True,
 u'NCPU': 8,
 u'NEventsListener': 0,
 u'NFd': 23,
 u'NGoroutines': 46,
 u'Name': u'fs0f642',
 u'NoProxy': u'',
 u'OSType': u'linux',
 u'OomKillDisable': True,
 u'OperatingSystem': u'Ubuntu 16.04 LTS',
 u'Plugins': {u'Authorization': None,
              u'Network': [u'overlay', u'host', u'bridge', u'null'],
              u'Volume': [u'local']},
 u'RegistryConfig': {u'IndexConfigs': {u'docker.io': {u'Mirrors': None,
                                                      u'Name': u'docker.io',
                                                      u'Official': True,
                                                      u'Secure': True},
                                       u'fs0f642.dibs.dk:5000': {u'Mirrors': [],
                                                                 u'Name': u'fs0f642.dibs.dk:5000',
                                                                 u'Official': False,
                                                                 u'Secure': False},
                                       u'fs0f642:5000': {u'Mirrors': [],
                                                         u'Name': u'fs0f642:5000',
                                                         u'Official': False,
                                                         u'Secure': False}},
                     u'InsecureRegistryCIDRs': [u'127.0.0.0/8'],
                     u'Mirrors': None},
 u'Runtimes': {u'runc': {u'path': u'docker-runc'}},
 u'SecurityOptions': [u'apparmor', u'seccomp'],
 u'ServerVersion': u'1.12.3',
 u'SwapLimit': False,
 u'Swarm': {u'Cluster': {u'CreatedAt': u'0001-01-01T00:00:00Z',
                         u'ID': u'',
                         u'Spec': {u'CAConfig': {},
                                   u'Dispatcher': {},
                                   u'Orchestration': {},
                                   u'Raft': {},
                                   u'TaskDefaults': {}},
                         u'UpdatedAt': u'0001-01-01T00:00:00Z',
                         u'Version': {}},
            u'ControlAvailable': False,
            u'Error': u'',
            u'LocalNodeState': u'inactive',
            u'Managers': 0,
            u'NodeAddr': u'',
            u'NodeID': u'',
            u'Nodes': 0,
            u'RemoteManagers': None},
 u'SystemStatus': None,
 u'SystemTime': u'2016-11-23T14:13:28.512139504+01:00'}
{u'ApiVersion': u'1.24',
 u'Arch': u'amd64',
 u'BuildTime': u'2016-10-26T22:01:48.986273588+00:00',
 u'GitCommit': u'6b644ec',
 u'GoVersion': u'go1.6.3',
 u'KernelVersion': u'4.4.0-45-generic',
 u'Os': u'linux',
 u'Version': u'1.12.3'}

```
##### SUMMARY

Getting started on https://github.com/ansible/ansible-container
don't work

##### STEPS TO REPRODUCE

Install ansible-container, and go through the current startup steps. And *Bang* the container failed.

```
ansible-container run
```

##### EXPECTED RESULTS

Expected: Getting started demo works, when hitting External: http://172.17.0.6:8080 in a browser.

##### ACTUAL RESULTS

Actual: Demo don't work.

```
Output in console:
gulp_1               | [BS] Access URLs:
gulp_1               |  -----------------------------------
gulp_1               |        Local: http://localhost:8080
gulp_1               |     External: http://172.17.0.6:8080
gulp_1               |  -----------------------------------
gulp_1               |           UI: http://localhost:3001
gulp_1               |  UI External: http://172.17.0.6:3001
gulp_1               |  -----------------------------------
gulp_1               | [BS] Couldn't open browser (if you are using BrowserSync in a headless environment, you might want to set the open option to false)
gulp_1               | { [Error: ENOENT: no such file or directory, open '/node/dist/maps/style.css.map']
gulp_1               |   errno: -2,
gulp_1               |   code: 'ENOENT',
gulp_1               |   syscall: 'open',
gulp_1               |   path: '/node/dist/maps/style.css.map' }
gulp_1               | [12:59:33] Finished 'sass' after 353 ms
gulp_1               | [12:59:33] Finished 'html' after 228 ms
django_1             | Traceback (most recent call last):
django_1             |   File "/usr/lib64/python2.7/wsgiref/handlers.py", line 85, in run
django_1             |     self.result = application(self.environ, self.start_response)
django_1             |   File "/venv/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
django_1             |     return self.application(environ, start_response)
django_1             |   File "/venv/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 189, in __call__
django_1             |     response = self.get_response(request)
django_1             |   File "/venv/lib/python2.7/site-packages/django/core/handlers/base.py", line 207, in get_response
django_1             |     return debug.technical_500_response(request, *sys.exc_info(), status_code=400)
django_1             |   File "/venv/lib/python2.7/site-packages/django/views/debug.py", line 97, in technical_500_response
django_1             |     html = reporter.get_traceback_html()
django_1             |   File "/venv/lib/python2.7/site-packages/django/views/debug.py", line 384, in get_traceback_html
django_1             |     return t.render(c)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 210, in render
django_1             |     return self._render(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 202, in _render
django_1             |     return self.nodelist.render(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 905, in render
django_1             |     bit = self.render_node(node, context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/debug.py", line 79, in render_node
django_1             |     return node.render(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/defaulttags.py", line 329, in render
django_1             |     return nodelist.render(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 905, in render
django_1             |     bit = self.render_node(node, context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/debug.py", line 79, in render_node
django_1             |     return node.render(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/debug.py", line 89, in render
django_1             |     output = self.filter_expression.resolve(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 648, in resolve
django_1             |     obj = self.var.resolve(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 789, in resolve
django_1             |     value = self._resolve_lookup(context)
django_1             |   File "/venv/lib/python2.7/site-packages/django/template/base.py", line 849, in _resolve_lookup
django_1             |     current = current()
django_1             |   File "/venv/lib/python2.7/site-packages/django/http/request.py", line 152, in build_absolute_uri
django_1             |     host=self.get_host(),
django_1             |   File "/venv/lib/python2.7/site-packages/django/http/request.py", line 102, in get_host
django_1             |     raise DisallowedHost(msg)
django_1             | DisallowedHost: Invalid HTTP_HOST header: 'django:8080'. You may need to add u'django' to ALLOWED_HOSTS.
django_1             | [23/Nov/2016 13:00:50] "GET / HTTP/1.1" 500 59

```

**Comment by chouseknecht** [comment]
@mkboel I experienced this yesterday too. Working on some changes and updates, so didn't get back to fixing this. Apologies. 

I did actually fix my fork of the example app. If you change line 16 in your settings.py file to look like [this](https://github.com/chouseknecht/django-gulp-nginx/blob/master/project/settings.py#L16), by adding 'localhost' and 'django' to ALLOWED_HOSTS, it *should* work.

**Comment by renoirb** [comment]
+1

**Comment by chouseknecht** [comment]
This has been working for a few weeks. But just to confirm it, I ran through the steps outlined in [Getting Started - Real world usage](http://docs.ansible.com/ansible-container/getting_started.html#real-world-usage-starting-from-a-working-base-setup), and I was able to run the following without any issues:

Using Docker for Mac from the Beta Channel on OSX 10.11.6

```
$ mkdir started 
$ cd started
$ ansible-container init ansible.django-gulp-nginx
$ ansible-container build
```

The build resulted in the following:
```
ansible-container_1  | PLAY RECAP *********************************************************************
ansible-container_1  | django                     : ok=14   changed=14   unreachable=0    failed=0
ansible-container_1  | gulp                       : ok=23   changed=19   unreachable=0    failed=0
ansible-container_1  | nginx                      : ok=17   changed=14   unreachable=0    failed=0
ansible-container_1  |
ansible_ansible-container_1 exited with code 0
Aborting on container exit...
Stopping ansible_nginx_1 ... done
Stopping ansible_django_1 ... done
Stopping ansible_postgresql_1 ... done
Stopping ansible_gulp_1 ... done
Exporting built containers as images...
Committing image...
Exported started-gulp with image ID sha256:d3c84071b8b638891aaae2aed3f546d9bbb801d60e79f8b0cafee02aac818a7a
Cleaning up gulp build container...
Committing image...
Exported started-nginx with image ID sha256:06232473d669aa8bf7bace6dff80f10445aba85b7ba55d5f1f5d5736330648a5
Cleaning up nginx build container...
Committing image...
Exported started-django with image ID sha256:91c887ded4f78022838cfab753e559696ccbdcf9a1e8422be0d38d894c3dec40
Cleaning up django build container...
Cleaning up Ansible Container builder...
```

And then running the application:

```
$ ansible-container run
No DOCKER_HOST environment variable found. Assuming UNIX socket at /var/run/docker.sock
Attaching to ansible_ansible-container_1
Cleaning up Ansible Container builder...
Attaching to ansible_postgresql_1, ansible_django_1, ansible_gulp_1, ansible_nginx_1
postgresql_1         | + export HOME=/var/lib/pgsql
postgresql_1         | + HOME=/var/lib/pgsql
django_1             | + [[ /usr/bin/dumb-init /venv/bin/python manage.py runserver 0.0.0.0:8080 == *\g\u\n\i\c\o\r\n* ]]
django_1             | + [[ /usr/bin/dumb-init /venv/bin/python manage.py runserver 0.0.0.0:8080 == *\r\u\n\s\e\r\v\e\r* ]]
django_1             | + '[' -f /django/manage.py ']'
postgresql_1         | + generate_passwd_file
django_1             | + /usr/bin/wait_on_postgres.py
postgresql_1         | ++ id -u
postgresql_1         | + export USER_ID=1000
django_1             | Waiting on postgresql to start...
postgresql_1         | + USER_ID=1000
django_1             | Postgresql started!
postgresql_1         | ++ id -g
django_1             | + '[' 0 == 0 ']'
django_1             | + /venv/bin/python manage.py migrate --fake-initial --noinput
django_1             | Operations to perform:
django_1             |   Synchronize unmigrated apps: staticfiles, messages, gunicorn
django_1             |   Apply all migrations: admin, contenttypes, auth, sessions
postgresql_1         | + export GROUP_ID=0
django_1             | Synchronizing apps without migrations:
postgresql_1         | + GROUP_ID=0
django_1             |   Creating tables...
postgresql_1         | + grep -v '^postgres' /etc/passwd
django_1             |     Running deferred SQL...
django_1             |   Installing custom SQL...
django_1             | Running migrations:
django_1             |   No migrations to apply.
postgresql_1         | + echo 'postgres:x:1000:0:PostgreSQL Server:/var/lib/pgsql:/bin/bash'
postgresql_1         | + export LD_PRELOAD=libnss_wrapper.so
django_1             | + exec /usr/bin/dumb-init /venv/bin/python manage.py runserver 0.0.0.0:8080
postgresql_1         | + LD_PRELOAD=libnss_wrapper.so
postgresql_1         | + export NSS_WRAPPER_PASSWD=/var/lib/pgsql/passwd
postgresql_1         | + NSS_WRAPPER_PASSWD=/var/lib/pgsql/passwd
postgresql_1         | + export NSS_WRAPPER_GROUP=/etc/group
postgresql_1         | + NSS_WRAPPER_GROUP=/etc/group
postgresql_1         | + whoami
postgresql_1         | + '[' '!' -f /var/lib/pgsql/data/userdata/pg_hba.conf ']'
postgresql_1         | postgres
postgresql_1         | + postgres
ansible_nginx_1 exited with code 1
gulp_1               | [23:05:33] Using gulpfile /node/gulpfile.js
gulp_1               | [23:05:33] Starting 'clean'...
gulp_1               | [23:05:33] Starting 'server'...
gulp_1               | [HPM] Proxy created: /admin  ->  http://django:8080
gulp_1               | [HPM] Proxy created: /static  ->  http://django:8080
gulp_1               | [HPM] Proxy created: /api  ->  http://django:8080
gulp_1               | [23:05:33] Finished 'server' after 24 ms
gulp_1               | [23:05:33] Starting 'watch'...
gulp_1               | [23:05:33] Finished 'watch' after 62 ms
gulp_1               | [23:05:33] Finished 'clean' after 116 ms
gulp_1               | [23:05:33] Starting 'build'...
gulp_1               | [23:05:33] Starting 'images'...
gulp_1               | [23:05:33] Starting 'templates'...
gulp_1               | [23:05:33] Starting 'sass'...
gulp_1               | [23:05:33] Starting 'clean-js'...
gulp_1               | [23:05:33] Starting 'fonts'...
gulp_1               | [23:05:33] Starting 'videos'...
gulp_1               | [23:05:33] Starting 'favicon'...
gulp_1               | [23:05:33] Starting 'lib'...
gulp_1               | [23:05:33] Finished 'build' after 38 ms
gulp_1               | [23:05:33] Starting 'default'...
gulp_1               | [23:05:33] Finished 'default' after 32 μs
gulp_1               | [23:05:33] gulp-imagemin: Minified 0 images
gulp_1               | [23:05:33] Finished 'images' after 48 ms
gulp_1               | [23:05:33] Starting 'html'...
gulp_1               | [23:05:33] Finished 'templates' after 34 ms
gulp_1               | [23:05:33] Finished 'fonts' after 26 ms
gulp_1               | [23:05:33] Finished 'videos' after 26 ms
gulp_1               | [23:05:33] Finished 'favicon' after 25 ms
gulp_1               | [23:05:33] Finished 'lib' after 23 ms
gulp_1               | [BS] Access URLs:
gulp_1               |  -----------------------------------
gulp_1               |        Local: http://localhost:8080
gulp_1               |     External: http://172.18.0.4:8080
gulp_1               |  -----------------------------------
gulp_1               |           UI: http://localhost:3001
gulp_1               |  UI External: http://172.18.0.4:3001
gulp_1               |  -----------------------------------
gulp_1               | [BS] Serving files from: dist
gulp_1               | [23:05:33] Finished 'clean-js' after 135 ms
gulp_1               | [23:05:33] Starting 'js'...
gulp_1               | [23:05:34] Finished 'js' after 290 ms
gulp_1               | [23:05:34] Finished 'sass' after 441 ms
gulp_1               | [23:05:34] Finished 'html' after 418 ms
```

And I'ma able to open http://localhost:8080 in Chrome without any issues.
